app/src/main/java/com/example/movie_list/MainActivity.kt

```diff
@@ -1,12 +1,85 @@
 package com.example.movie_list
 
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import android.widget.LinearLayout
+import com.example.movie_list.actions.ActionCreator
+import com.example.movie_list.model.AppState
+import com.example.movie_list.model.Movie.MovieItem
+import com.example.movie_list.model.Movie.movies
+import com.github.raulccabreu.redukt.states.StateListener
+import trikita.anvil.DSL.*
+import trikita.anvil.RenderableView
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), StateListener<AppState> {
+
+    protected val state: AppState
+        get() = MovieListApp.redukt.state
+
+    override fun onStart() {
+        super.onStart()
+        MovieListApp.redukt.listeners.add(this)
+        onChanged(state)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        MovieListApp.redukt.listeners.remove(this)
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+
+        movies.add(MovieItem("title1","2010","poster1","comedy1"))
+        movies.add(MovieItem("title2","2011","poster2","comedy2"))
+        movies.add(MovieItem("title3","2012","poster3","comedy3"))
+        movies.add(MovieItem("title4","2013","poster4","comedy4"))
+        movies.add(MovieItem("title5", "2041", "poster5", "comedy5"))
+        movies.add(MovieItem("title6","2016","poster6","comedy6"))
+        movies.add(MovieItem("title7","2015","poster7","comedy7"))
+
+        setContentView(getView())
+    }
+
+    override fun hasChanged(newState: AppState, oldState: AppState): Boolean {
+        if (newState.list != oldState.list) return true
+        return false
+    }
+
+    override fun onChanged(state: AppState) {
+        if(state.list!=null) {

```

Comment:
This is just a suggestion of syntax only to show you some alternatives that we have in Kotlin. The code inside the `let` will be executed **only** if state.list would be different than null.
```suggestion
        state.list?.let{...}
```